Content.Server/Medical/HealthAnalyzerSystem.cs

```diff
@@ -59,15 +110,99 @@ private void OpenUserInterface(EntityUid user, EntityUid analyzer)
             if (!TryComp<ActorComponent>(user, out var actor) || !_uiSystem.TryGetUi(analyzer, HealthAnalyzerUiKey.Key, out var ui))
                 return;
 
-            _uiSystem.OpenUi(ui ,actor.PlayerSession);
+            _uiSystem.OpenUi(ui, actor.PlayerSession);
         }
 
-        public void UpdateScannedUser(EntityUid uid, EntityUid user, EntityUid? target, HealthAnalyzerComponent? healthAnalyzer)
+        /// <summary>
+        /// Mark the entity as having its health analyzed, and link the analyzer to it
+        /// </summary>
+        private void BeginAnalyzingEntity(EntityUid uid, EntityUid healthAnalyzerUid, HealthAnalyzerComponent? component = null, HealthBeingAnalyzedComponent? healthBeingAnalyzedComponent = null)
         {
-            if (!Resolve(uid, ref healthAnalyzer))
+            if (!Resolve(healthAnalyzerUid, ref component))
                 return;
 
-            if (target == null || !_uiSystem.TryGetUi(uid, HealthAnalyzerUiKey.Key, out var ui))
+            if (!Resolve(uid, ref healthBeingAnalyzedComponent, false))
+            {
+                //No other active analyzers, create the component and attach to the scanned entity
+                healthBeingAnalyzedComponent = new HealthBeingAnalyzedComponent();
+                AddComp(uid, healthBeingAnalyzedComponent);
+            }
```

Comment:
```suggestion
var healthBeingAnalyzedComponent = EnsureComp<HealthBeingAnalyzedComponent>(uid);
```
hence you can remove the healthAnalyzer param

Reply:
Done, forgot this exists